app/src/main/java/com/github/se/studybuddies/MainActivity.kt

```diff
@@ -99,7 +99,17 @@ class MainActivity : ComponentActivity() {
           NavHost(navController = navController, startDestination = startDestination) {
             composable(Route.START) {
               if (auth.currentUser != null) {
-                navController.navigate(Route.SOLOSTUDYHOME)
+
+                db.userExists(
+                    uid = db.getCurrentUserUID(),
+                    onSuccess = { userExists ->
+                      if (userExists) {
+                        navController.navigate(Route.SOLOSTUDYHOME)
+                      } else {
+                        navController.navigate(Route.CREATEACCOUNT)
+                      }
+                    },
+                    onFailure = { Log.d("MyPrint", "Failed to check user existence") })
```

Comment:
Nice check, however what happen if the the call to the database is a failure, we got a Log but we block the user.

It's a big issue because we are making the offline functionality unusable without internet. The user is blocked into a blue screen !

Additionally, it will be cleaner if you move your check into UserViewModel and then call the viewModel.

Reply:
I solved the offline issue, it should be good now. However, for the viewModel it just seems very complicated for such a simple thing, especially since the database is already instanciated in the main and I'm just calling its functions.